### PR TITLE
Prototype of Angular running atop Keystone

### PR DESCRIPTION
**Pull request by tremainebuchanan**
This branch was created with the view of testing the usage of Angular atop Keystone to build out the SRF dynamic site. 

The code is now being merged to facilitate further build out as the test/experiment worked well.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "license": "MIT",
+  "ignore": [
+    "**/.*",
+    "node_modules",
+    "bower_components",
+    "test",
+    "tests"
+  ],
+  "dependencies": {
+    "angular": "~1.4.8",
```
Comment (matjames007) [inline bower.json:19]: Just something to point out... we should be building against a fixed version of angular and other dependencies.  `~` is an instruction to pull a version reasonably close to the version specified.

File: models/Menu.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+	var Menu = new keystone.List('Menu', {
+		map: { name: 'label' },
+		autokey: { path: 'slug', from: 'label', unique: true }
+	});
+
+	Menu.add({
+		label: {type: String, initial: true, required: true, unique: true},
+		link: {type: String, initial: true, required: true, unique: true},
+		order: {type: Types.Select, options: '0, 1, 2, 3, 4', required: true, default: '0'},
```
Comment (matjames007) [inline models/Menu.js:13]: Why limit this to the range `{0 ... 4}`?

File: public/app/modules/home/home.controller.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * Created by tremaine on 1/15/16.
+ */
+(function(){
+    'use strict';
+    angular
+        .module('home')
+        .controller('Home', Home);
+
+    Home.$inject = ['$scope', 'dataService'];
+    /**
```
Comment (matjames007) [inline public/app/modules/home/home.controller.js:11]: Key to good coding is documentation.  Haven't seen any so far! @tremainebuchanan

File: bower.json (hunk 1 above)
Comment (tremainebuchanan) [inline bower.json:19]: Noted. This was changed in the version I'm working on right now.

File: public/app/modules/home/home.controller.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        activate();
+
+        function activate(){
+            dataService.getPosts().then(function(data){
+                console.log(data);
+            });
+        }
+
+        $scope.slides = [{
```
Comment (matjames007) [inline public/app/modules/home/home.controller.js:27]: @tremainebuchanan please change this from being hardcoded to something configurable.  Doesn't have to get done right away, but remember the importance of it.  Open an issue with this.

File: public/app/modules/home/home.html [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+<div class="container">
+	<div class="row">
+		<div class="col-sm-12 col-md-4 col-lg-4">
+			<div class="row">
+				<div class="col-sm-6 col-md-12 col-lg-12">
+					<div class="box box-4col box-1" style="background-image: url(/images/post_its.jpg);">
+						<div class="box-caption">
+							<div class="label label-text-2 black-text">IDEAS</div>
+                                    <span class="title-2 white-text">
+                                        A Caribbean Open Data Public Service
```
Comment (matjames007) [inline public/app/modules/home/home.html:18]: @tremainebuchanan I am guessing all of the fixed/hardcoded stuff will be changed?

File: public/app/modules/home/home.html [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	<div class="row" ng-include="'app/modules/partials/carousel.html'">
+
+	</div>
+</div>
+<div class="container">
+	<div class="row">
+		<div class="col-sm-12 col-md-4 col-lg-4">
+			<div class="row">
+				<div class="col-sm-6 col-md-12 col-lg-12">
+					<div class="box box-4col box-1" style="background-image: url(/images/post_its.jpg);">
```
Comment (matjames007) [inline public/app/modules/home/home.html:14]: @tremainebuchanan separation of concerns... styling belongs in the stylesheet!

File: public/app/modules/partials/navbar.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+							<span class="caret"></span>
+						</a>
+						<ul class="dropdown-menu" uib-dropdown-menu>
+							<li><a href="#/projects">Projects</a></li>
+							<li><a href="#/initiatives">Initiatives</a></li>
+							<li><a href="#/community">Community</a></li>
+							<li><a href="#/our-process">Our Process</a></li>
+						</ul>
+					</li>
+					<li><a class="dark-grey-text" href="#/ideas">Ideas</a></li>
```
Comment (matjames007) [inline public/app/modules/partials/navbar.html:27]: @tremainebuchanan shouldn't these links be pulled in dynamically?  What if we want to add in another menu?

File: public/app/modules/shared_services/dataservice.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+    dataService.$inject = ['$http'];
+    function dataService($http){
+        var service = {
+            getPosts : getPosts
+        };
+
+        return service;
+
+        function getPosts(){
```
Comment (matjames007) [inline public/app/modules/shared_services/dataservice.js:17]: @tremainebuchanan if we are using the angular framework, let's use it properly.  Don't use `$http.get()` - use the factory method angular specifies.

File: public/bower_components/angular-bootstrap/.bower.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (matjames007) [inline public/bower_components/angular-bootstrap/.bower.json:1]: @tremainebuchanan exclude the `bower_components` directory from git using the `.gitignore` file.

**Comment by matjames007** [comment]
@tremainebuchanan the review process breaksdown if the `bower_components` folder is included.  Github realizes there is too many files and hides them as a result.  Try and exclude this and push the changes.  I am sure there are other files that I have missed during my review process.

**Pull request merged**